### PR TITLE
Add Duration to Config and registerEntity (#47)

### LINKED ISSUE #47: Allow override of maxSessionIdleTime per entity
Either by config or on registerEntity.

I have many entities that don't need to live long, they are called once and then will lie dormant for quite a long time. On the other hand there are some that are called with a long call timeout because it has to deal with slow interfaces, but increasing globally the idleTimeout for that uses a lot of memory to hold the short-lived ones.

I have solution already, quite simple to do

### PR DESCRIPTION
**Pull request by csar**
prefer registerEntity, the config extension, then global default

Closes #47

### CODE REVIEW

File: entities/src/main/scala/com/devsisters/shardcake/Config.scala [hunk 1]
```diff
@@ -29,7 +29,8 @@ case class Config(
   sendTimeout: Duration,
   refreshAssignmentsRetryInterval: Duration,
   unhealthyPodReportInterval: Duration,
-  simulateRemotePods: Boolean
+  simulateRemotePods: Boolean,
+  entityMaxIdleTimes : Map[String,Duration]
```
Comment (ghostdogpr) [inline entities/src/main/scala/com/devsisters/shardcake/Config.scala]: I think that one is not necessary if you have the parameter in `registerEntity`, it makes too many ways of doing the same thing.

File: entities/src/main/scala/com/devsisters/shardcake/Sharding.scala [hunk 2]
```diff
@@ -311,15 +311,16 @@ class Sharding private (
...
+    entityMaxIdleTime  :Option[Duration] = None
+  ): URIO[Scope with R, Unit] = registerRecipient(entityType, behavior, terminateMessage, entityMaxIdleTime) *>
     eventsHub.publish(ShardingRegistrationEvent.EntityRegistered(entityType)).unit
 
   def registerTopic[R, Req: Tag](
     topicType: TopicType[Req],
     behavior: (String, Dequeue[Req]) => RIO[R, Nothing],
     terminateMessage: Promise[Nothing, Unit] => Option[Req] = (_: Promise[Nothing, Unit]) => None
-  ): URIO[Scope with R, Unit] = registerRecipient(topicType, behavior, terminateMessage) *>
+  ): URIO[Scope with R, Unit] = registerRecipient(topicType, behavior, terminateMessage, Some(Duration.Infinity)) *>
```
Comment (ghostdogpr) [inline entities/src/main/scala/com/devsisters/shardcake/Sharding.scala]: I don't think we want infinity here, it's the same logic as entities.

**Review by ghostdogpr — COMMENTED** [review]
Can you also run `fmt` in sbt to format the code? 🙏

File: entities/src/main/scala/com/devsisters/shardcake/Config.scala (hunk 1 above)
Comment (csar) [reply entities/src/main/scala/com/devsisters/shardcake/Config.scala]: What do you prefer - if we gp config only it becomes much simpler. then the only change will be in startExpirationFiber

Comment (ghostdogpr) [reply entities/src/main/scala/com/devsisters/shardcake/Config.scala]: I think I prefer the parameter in `registerEntity` actually, the map feels a little awkward using Strings as keys. In `registerEntity` I find it clearer.

File: entities/src/main/scala/com/devsisters/shardcake/Sharding.scala (hunk 2 above)
Comment (csar) [reply entities/src/main/scala/com/devsisters/shardcake/Sharding.scala]: I agree, it should be None

File: entities/src/main/scala/com/devsisters/shardcake/Config.scala (hunk 1 above)
Comment (csar) [reply entities/src/main/scala/com/devsisters/shardcake/Config.scala]: OK, I remove the change to Config

**Review by ghostdogpr — APPROVED** [review]
LGTM!

Can you submit the same PR for the `series/1.x` branch? I keep both of them with the same features for now. Thanks!

**Pull request merged**